sql-data-pipeline/src/main/java/io/cdap/pipeline/sql/app/bigquery/BigQueryExecutor.java

```diff
@@ -0,0 +1,332 @@
+import io.cdap.pipeline.sql.api.template.SQLSource;
+import io.cdap.pipeline.sql.api.template.SQLTransform;
+import io.cdap.pipeline.sql.api.template.tables.AbstractTableInfo;
+import io.cdap.pipeline.sql.app.core.SQLConfig;
+import io.cdap.pipeline.sql.app.core.TemporaryTable;
+import org.apache.calcite.jdbc.CalciteSchema;
+import org.apache.calcite.jdbc.Driver;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.rel2sql.RelToSqlConverter;
+import org.apache.calcite.rel.type.RelDataTypeField;
+import org.apache.calcite.schema.ColumnStrategy;
+import org.apache.calcite.schema.SchemaPlus;
+import org.apache.calcite.sql.SqlDataTypeSpec;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.ddl.SqlCreateTable;
+import org.apache.calcite.sql.ddl.SqlDdlNodes;
+import org.apache.calcite.sql.dialect.BigQuerySqlDialect;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.sql.type.SqlTypeUtil;
+import org.apache.calcite.sql.util.SqlBuilder;
+import org.apache.calcite.tools.FrameworkConfig;
+import org.apache.calcite.tools.Frameworks;
+import org.apache.calcite.tools.RelBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A custom action which executes SQL upon a target platform.
+ */
+public class BigQueryExecutor implements CustomAction {
+  private static final String TEMPORARY_TABLE_PREFIX = "temporary_table_";
+  private static final String CONFIG_NAME = "serializedSqlConfig";
+  private static final String SERVICE_ACCOUNT_PATH_NAME = "serviceAccountPath";
+
+  private SQLConfig config;
+  private Dag dag;
+  private String serviceAccountPath;
+  private Set<String> sourceStages;
+  private Set<String> sinkStages;
+  private Map<String, SQLTransform> pluginMap;
+  private Map<String, Set<String>> fromNodeMap;
+  private Map<String, Set<String>> toNodeMap;
+  private Map<String, RelNode> relationalNodeMap;
+  private Map<String, String> temporaryTableMap;
+  private List<String> queries;
+
+  private static final Logger LOG = LoggerFactory.getLogger(BigQueryExecutor.class);
+
+  public BigQueryExecutor(SQLConfig config, Map<String, SQLTransform> pluginMap) {
+    this.config = config;
+    this.sourceStages = new HashSet<>();
+    this.sinkStages = new HashSet<>();
+    this.pluginMap = pluginMap;
+    this.fromNodeMap = new HashMap<>();
+    this.toNodeMap = new HashMap<>();
+    this.relationalNodeMap = new HashMap<>();
+    this.temporaryTableMap = new HashMap<>();
+    this.queries = new ArrayList<>();
+  }
+
+  @Override
+  public void configure(CustomActionConfigurer configurer) {
+    // Initialize the query generation
+    init();
```

Comment:
Isn't it a good idea to perform validation in the configuration stage so that invalid pipelines fail fast? The init() logic also computes and orders the queries as well.